### PR TITLE
Fix MIGraphX padding to TOSA padding

### PR DESCRIPTION
**Pull request by manupak**
Currently MIGraphX provides padding as
[hlow, wlow, hhigh, whigh] while TOSA expects
[hlow, hhigh, wlow, whigh]. However, our code
just copies it as it is.

This commit fixes this problem.
closes : https://github.com/ROCmSoftwarePlatform/rocMLIR-internal/issues/1233

### CODE REVIEW

File: mlir/test/fusion/pr-e2e/mixr-slice-convolution-add-small.mlir [hunk 1]
```diff
@@ -0,0 +1,9 @@
+// RUN: rocmlir-gen -fut mlir_slice_convolution_add --arch %arch --clone-harness %s | rocmlir-driver -kernel-pipeline=migraphx | rocmlir-driver -host-pipeline=migraphx,highlevel | rocmlir-gen -ph -rand 1 -rand_type float -fut mlir_slice_convolution_add_wrapper --verifier clone -relDiff_threshold 0.000002 - | rocmlir-driver -host-pipeline mhal -kernel-pipeline full | xmir-runner --shared-libs=%linalg_test_lib_dir/libmlir_rocm_runtime%shlibext,%conv_validation_wrapper_library_dir/libconv-validation-wrappers%shlibext,%linalg_test_lib_dir/libmlir_runner_utils%shlibext,%linalg_test_lib_dir/libmlir_float16_utils%shlibext,%linalg_test_lib_dir/libmlir_c_runner_utils%shlibext,%linalg_test_lib_dir/libmlir_async_runtime%shlibext --entry-point-result=void | FileCheck %s
+// ALLOW_RETRIES: 2
+// CHECK: [1 1 1]
+func.func private @mlir_slice_convolution_add(%arg0: tensor<1x2x4x4xf32>, %arg1: tensor<1x8x4x4xf32>, %arg2: tensor<2x2x1x7xf32>) -> tensor<1x2x4x4xf32> attributes {arch = "gfx908:sramecc+:xnack-", kernel = "mixr", num_cu = 120 : i64} {
```
Comment (krzysz00) [inline mlir/test/fusion/pr-e2e/mixr-slice-convolution-add-small.mlir]: We probably want to take the `arch` off of this?

File: mlir/test/fusion/nightly-misc-e2e/mixr-slice-convolution-add.mlir [hunk 2]
```diff
@@ -0,0 +1,9 @@
+// RUN: rocmlir-gen -fut mlir_slice_convolution_add --arch %arch --clone-harness %s | rocmlir-driver -kernel-pipeline=migraphx | rocmlir-driver -host-pipeline=migraphx,highlevel | rocmlir-gen -ph -rand 1 -rand_type float -fut mlir_slice_convolution_add_wrapper --verifier clone -relDiff_threshold 0.000002 - | rocmlir-driver -host-pipeline mhal -kernel-pipeline full | xmir-runner --shared-libs=%linalg_test_lib_dir/libmlir_rocm_runtime%shlibext,%conv_validation_wrapper_library_dir/libconv-validation-wrappers%shlibext,%linalg_test_lib_dir/libmlir_runner_utils%shlibext,%linalg_test_lib_dir/libmlir_float16_utils%shlibext,%linalg_test_lib_dir/libmlir_c_runner_utils%shlibext,%linalg_test_lib_dir/libmlir_async_runtime%shlibext --entry-point-result=void | FileCheck %s
+// ALLOW_RETRIES: 2
+// CHECK: [1 1 1]
+func.func private @mlir_slice_convolution_add(%arg0: tensor<1x64x73x73xf32>, %arg1: tensor<1x128x73x73xf32>, %arg2: tensor<64x64x1x7xf32>) -> tensor<1x64x73x73xf32> attributes {arch = "gfx908:sramecc+:xnack-", kernel = "mixr", num_cu = 120 : i64} {
```
Comment (krzysz00) [inline mlir/test/fusion/nightly-misc-e2e/mixr-slice-convolution-add.mlir]: Same here

**Review by krzysz00 — APPROVED** [review]
Approved except that I don't know if that test will work right outside of MI-100 (or, if it does' it'll work by accident)

And I'll approve the backport

File: mlir/test/fusion/pr-e2e/mixr-slice-convolution-add-small.mlir (hunk 1 above)
Comment (manupak) [reply mlir/test/fusion/pr-e2e/mixr-slice-convolution-add-small.mlir]: done

File: mlir/test/fusion/nightly-misc-e2e/mixr-slice-convolution-add.mlir (hunk 2 above)
Comment (manupak) [reply mlir/test/fusion/nightly-misc-e2e/mixr-slice-convolution-add.mlir]: done

**Comment by manupak** [comment]
removed those now! thanks for catching it

**Review by jerryyin — APPROVED** [review]
Good job!

**Review by umangyadav — APPROVED** [review]
Looks good logically. 
Has this been tested with small reproducer and inceptionNet to see if it works ? 
I am building this branch with MIGraphX, may take a while for me to verify.

**Review by ahsan-ca — APPROVED** [review]

**Pull request merged**